src/components/transfers/CreateForm.tsx

```diff
@@ -0,0 +1,235 @@
+  approvedById: yup.string().uuid().notRequired().nullable(),
+  sourceVaultId: yup.string().uuid().required(),
+  sourceCampaignId: yup.string().uuid(),
+  targetVaultId: yup.string().uuid().required(),
+  targetCampaignId: yup.string().uuid(),
+})
+
+export default function CreateForm() {
+  const { t } = useTranslation('transfer')
+  const router = useRouter()
+
+  const { data: campaigns } = useCampaignList()
+  const { data: vaults } = useVaultsList()
+  const { data: people } = usePersonList()
+
+  const [sourceCampaignId, setSourceCampaignId] = useState('')
+  const [targetCampaignId, setTargetCampaignId] = useState('')
+
+  const initialValues: TransferInput = {
+    status: TransferStatus.initial,
+    currency: '',
+    amount: 0,
+    reason: '',
+    documentId: '',
+    targetDate: '',
+    approvedById: '',
+    sourceCampaignId: '',
+    sourceVaultId: '',
+    targetCampaignId: '',
+    targetVaultId: '',
+  }
+
+  const mutation = useMutation<
+    AxiosResponse<TransferResponse>,
+    AxiosError<ApiErrors>,
+    TransferInput
+  >({
+    mutationFn: useCreateTransfer(),
+    onError: () => AlertStore.show(t('alerts:error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('alerts:create'), 'success')
+      router.push(routes.admin.transfer.index)
+    },
+  })
+
+  function handleSubmit(values: TransferInput) {
+    const data: TransferInput = {
+      status: TransferStatus.initial,
+      currency: values.currency,
+      amount: values.amount,
+      reason: values.reason,
+      documentId: values.documentId ? values.documentId : null,
+      targetDate: values.targetDate ? new Date(values.targetDate) : null,
+      approvedById: values.approvedById ? values.approvedById : null,
+      sourceCampaignId: sourceCampaignId,
+      sourceVaultId: values.sourceVaultId,
+      targetCampaignId: targetCampaignId,
+      targetVaultId: values.targetVaultId,
+    }
+    if (!data.sourceCampaignId || !data.targetCampaignId) {
+      return
+    }
+    mutation.mutate(data)
+  }
+
+  return (
+    <GenericForm
+      onSubmit={handleSubmit}
+      initialValues={initialValues}
+      validationSchema={validationSchema}>
+      <Box>
+        <Typography variant="h5" component="h2" sx={{ marginBottom: 2, textAlign: 'center' }}>
+          {t('form-heading')}
+        </Typography>
+        <Grid container spacing={2} sx={{ width: 600, margin: '0 auto' }}>
+          <Grid item xs={12}>
+            <FormTextField type="string" label={t('reason')} name="reason" autoComplete="reason" />
+          </Grid>
+          <Grid item xs={12}>
+            <FormSelect name="currency" label={t('currency')}>
+              {(Object.values(Currency) || []).map((currency) => {
+                return (
+                  <MenuItem key={currency} value={currency}>
+                    {currency}
+                  </MenuItem>
+                )
+              })}
+            </FormSelect>
```

Comment:
Check with the team about a component `CurrencySelect`

Someone was working on it